### PR TITLE
5743 `assertTrue` ignores the label, if assigned to AssertionResult

### PR DESCRIPTION
**Pull request by jupposessho**

### CODE REVIEW

**Comment by kitlangton** [comment]
Oh! Nicely done 😄

**Comment by jupposessho** [comment]
> Oh! Nicely done 😄

I did my best :D

File: test-tests/shared/src/test/scala/zio/test/ReportingTestUtils.scala [hunk 1]
```diff
@@ -156,6 +156,13 @@ object ReportingTestUtils {
     withOffset(2)(assertSourceLocation() + "\n")
   )
 
+  def test9(implicit trace: ZTraceElement): ZSpec[Any, Nothing] = test("labeled failures") {
+    assertTrue(1 == 1).map(_.label("first")) &&
+    assertTrue(1 == 1).map(_.label("second")) &&
+    assertTrue(1 == 0).map(_.label("third")) &&
+    assertTrue(1 == 0).map(_.label("fourth"))
```
Comment (kitlangton) [inline test-tests/shared/src/test/scala/zio/test/ReportingTestUtils.scala:163]: Probably would be slightly more ergonomic if `label` could be called directly, such as `assertTrue(1 == 1).label("something")`. But the unification of these different assertion result types may make that implementation a bit painful. This is a great start though.

**Review by kitlangton — APPROVED** [review]
I'm going to follow up, seeing if I can better unify the APIs. But this is a lovely start.

**Comment by jupposessho** [comment]
> I'm going to follow up, seeing if I can better unify the APIs. But this is a lovely start.

Let me know if I can help with the follow up.

**Pull request merged**